java/algorithm/src/main/java/org/apache/arrow/algorithm/sort/VariableWidthOutOfPlaceVectorSorter.java

```diff
@@ -45,6 +46,22 @@ public void sortOutOfPlace(V srcVector, V dstVector, VectorValueComparator<V> co
     ArrowBuf dstValueBuffer = dstVector.getDataBuffer();
     ArrowBuf dstOffsetBuffer = dstVector.getOffsetBuffer();
 
+    // check buffer size
+    Preconditions.checkArgument(dstValidityBuffer.capacity() * 8 >= srcVector.getValueCount(),
+        "Not enough capacity for the validity buffer of the dst vector. " +
+            "Expected capacity %s, actual capacity %s",
+        (srcVector.getValueCount() + 7) / 8, dstValidityBuffer.capacity());
+    Preconditions.checkArgument(
+        dstOffsetBuffer.capacity() >= srcVector.getValueCount() * BaseVariableWidthVector.OFFSET_WIDTH,
+        "Not enough capacity for the offset buffer of the dst vector. " +
+            "Expected capacity %s, actual capacity %s",
+        srcVector.getValueCount() * BaseVariableWidthVector.OFFSET_WIDTH, dstValueBuffer.capacity());
```

Comment:
nit: `dstValueBuffer` -> `dstOffsetBuffer`

Reply:
Nice catch. Thank you.